### PR TITLE
Iterator

### PR DESCRIPTION
**Pull request by emhane**
Expose iterator over entries publicly

### CODE REVIEW

**Comment by vasilakisfil** [comment]
Just happened to need an expiring hashmap and took a look in your PR, so not really affiliated with the project. Maybe your PR is not completed yet, but if it is not sure what this adds since there is already the [keys](https://docs.rs/delay_map/latest/delay_map/hashmap_delay/struct.HashMapDelay.html#method.keys) method that does the same thing.

**Comment by emhane** [comment]
Well @vasilakisfil you seem to have missed the fact that I expose the iterator over the private field `entries` publicly. A method with the same name exists in `HashMapDelay`, check that out. But sure internally that `iter` method can use the keys method.

**Comment by vasilakisfil** [comment]
@emhane I think I missed the fact that you add the method in `HashSetDelay` while I was looking at `HashMapDelay`... looks good :+1:

**Review by AgeManning — APPROVED** [review]
Looks good. 

Thanks for the doc improvements and typo fix also.

File: src/hashset_delay.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-/// The default delay for entries, in seconds. This is only used when `insert()` is used to add
+/// The default delay for entries, in seconds. This is only used when [`HashSetDelay::default`] is
+/// used to instantiate the [`HashSetDelay`] and then [`HasSetDelay::insert`] is used to add
```
Comment (AgeManning) [inline src/hashset_delay.rs:2]:
```suggestion
/// used to instantiate the [`HashSetDelay`] and then [`HashSetDelay::insert`] is used to add
```

**Pull request merged**